presto-spark-base/src/main/java/com/facebook/presto/spark/execution/task/PrestoSparkNativeTaskExecutorFactory.java

```diff
@@ -503,7 +503,12 @@ public Tuple2<MutablePartitionId, T> next()
             checkArgument(outputType == PrestoSparkSerializedPage.class,
                     format("PrestoSparkNativeTaskExecutorFactory only outputType=PrestoSparkSerializedPage" +
                             "But tried to extract outputType=%s", outputType));
-            return new Tuple2<>(new MutablePartitionId(), (T) toPrestoSparkSerializedPage(next.get()));
+            MutablePartitionId mutablePartitionId = new MutablePartitionId();
```

Comment:
The `partitionId` is available where the constructor of this class is invoked. You could use that to avoid handling uncertainty with getting `TaskInfo` here